common/lib/xmodule/xmodule/capa_base.py

```diff
@@ -157,7 +158,12 @@ class CapaFields(object):
             {"display_name": _("Per Student"), "value": RANDOMIZATION.PER_STUDENT}
         ]
     )
-    data = String(help=_("XML data for the problem"), scope=Scope.content, default="<problem></problem>")
+    data = XMLString(
+        help=_("XML data for the problem"),
+        scope=Scope.content,
+        enforce_type=FEATURES.get('ENABLE_XBLOCK_XML_VALIDATION', True),
```

Comment:
Sadly, we don't have a current standard for a way to set field-level values at runtime through xblock-specific apis.